# HG changeset patch
# User Andreas Weber <andy.weber.aw@gmail.com>
# Date 1464851320 -7200
#      Thu Jun 02 09:08:40 2016 +0200
# Node ID 66972e1679df3150558c551824aade9d1669110e
# Parent  4acfe6a99a9f25b3ad8663beb29322d2cf0b3d5f
configure.ac: Fix check for OpenGL support in FLTK

FLTK needs C++ to compile so this check always failed whether
the OpenGL headers were present or not in FL.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1610,36 +1610,37 @@ if test $check_fltk = yes; then
     FLTK_CPPFLAGS=`$FLTK_CONFIG $fltkconf_args --use-gl --cflags`
     FLTK_LDFLAGS=`$FLTK_CONFIG $fltkconf_args --use-gl --ldflags`
 
     case $host_os in
       mingw*)
         FLTK_LDFLAGS=`echo $FLTK_LDFLAGS | $SED -e 's/-mwindows//g'`
       ;;
     esac
-
+    AC_LANG_PUSH(C++)
     AC_CACHE_CHECK([for OpenGL support in FLTK],
       [octave_cv_fltk_opengl_support],
       [save_CPPFLAGS="$CPPFLAGS"
       CPPFLAGS="$CFLAGS $FLTK_CPPFLAGS"
       AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
         #include <FL/gl.h>
         ]], [[
         int nothing = 0;
         ]])],
-        octave_cv_fltk_opengl_support=no,
-        octave_cv_fltk_opengl_support=yes)
+        octave_cv_fltk_opengl_support=yes,
+        octave_cv_fltk_opengl_support=no)
       CPPFLAGS="$save_CPPFLAGS"
       ])
     if test $octave_cv_fltk_opengl_support = no; then
       warn_fltk_opengl="FLTK does not have OpenGL support.  OpenGL graphics with FLTK widgets will be disabled."
     else
       build_fltk_graphics=yes
       AC_DEFINE(HAVE_FLTK, 1, [Define to 1 if FLTK is available.])
     fi
+    AC_LANG_POP(C++)
 
     if test -n "$warn_fltk_opengl"; then
       OCTAVE_CONFIGURE_WARNING([warn_fltk_opengl])
     fi
   fi
 fi
 
 AC_SUBST(FLTK_CPPFLAGS)
